lib/vix/vips/image.ex

```diff
@@ -323,15 +334,13 @@ defmodule Vix.Vips.Image do
   """
   @spec write_to_stream(__MODULE__.t(), String.t()) :: Enumerable.t()
   def write_to_stream(%Image{ref: vips_image}, suffix) do
-    require Logger
-
     Stream.resource(
       fn ->
-        {:ok, pipe} = Vix.Target.new(vips_image, suffix)
+        {:ok, pipe} = Vix.TargetPipe.new(vips_image, suffix)
         pipe
       end,
       fn pipe ->
-        ret = Vix.Target.read(pipe)
+        ret = Vix.TargetPipe.read(pipe)
```

Comment:
Would you consider adding a `size` option to `Vix.Vips.TargetPipe.read`?  When doing streaming uploads to S3, the minimum chunk size is 5 MiB. Being able to specify a chunk size would be very helpful in these situations.  Possibly the chunking could be pushed into the NIF itself but probably that's a bigger challenge?

Reply:
That would mean that the `size` would need to be specified as an argument to `write_to_stream/2` which would lead to a signature change I suppose.

Reply:
I'm aware of this case. I slightly more in favor of keeping NIF and genserver logic simple and handle buffer at elixir side. 

Something like:

```elixir
def buffer(stream, buffer_size) do
  Stream.chunk_while(
    stream,
    [],
    fn bin, acc ->
      acc_size = IO.iodata_length(acc)
  
      if IO.iodata_length(bin) + acc_size >= buffer_size do
        size = buffer_size - acc_size
        <<chunk::binary-size(size), rest::binary>> = bin
        {:cont, [chunk | acc], [rest]}
      else
        {:cont, [bin | acc]}
      end
    end,
    fn
      [] -> {:cont, []}
      acc -> {:cont, acc, []}
    end
  )
end

Image.write_to_stream(image, ".png")
|> buffer(5 * 1024 * 1024)
```

We can change `write_to_stream` to do this for the user. wdyt?

Reply:
Thats a great idea, and I started along a similar track after I wrote my original comment.  And there's even a good case that this doesn't belong in `Vix`; perhaps its better in a client library like `Image`?

Reply:
Agree. I was thinking along the same lines and skipped it. But it is bit blurry if this should be part of Vix. 

For the time being, I'll skip adding it and leave it to `Image` then.